[feat-19] Prefect로 작업 관리 및 모델 업데이트

lazychoi opened the pull request:
tasks.py
- db에서 데이터 불러오기
- 전처리: 불필요한 컬럼 삭제, 결측치 처리, 인코딩, 정규화
- 모델 학습
- mlflow로parameters, metrics, model 저장
- metric(auc)이 기존 production된 모델보다 높으면 교체

train_flow.py :
- 위 tasks를 실행

issue : #19

lazychoi committed:
[feat-19] Prefect로 작업 관리 및 모델 업데이트

tasks.py
- db에서 데이터 불러오기
- 전처리: 불필요한 컬럼 삭제, 결측치 처리, 인코딩, 정규화
- 모델 학습
- mlflow로parameters, metrics, model 저장
- metric(auc)이 기존 production된 모델보다 높으면 교체

train_flow.py :
- 위 tasks를 실행

issue : #19

pastnfast reviewed (approved):
LGTM

AstroNaut000 commented on train_prefect/tasks.py:
```diff
@@ -0,0 +1,168 @@
+@task(log_stdout=True, nout=2)
+def train_model(model, x, y):
+
+    x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, stratify=y)
```
x_train,**x_test**,y_train,**y_test**에서 test 부분을 val로 변경

Ilevk commented on train_prefect/train_flow.py:
```diff
@@ -0,0 +1,32 @@
+model = XGBClassifier(use_label_encoder=False, **params)
+model_name = model.__class__.__name__
+
+with Flow("Model Training Flow") as flow:
```
Flow 이름은 식별이 가능하도록 변경

Ilevk commented on train_prefect/train_flow.py:
```diff
@@ -0,0 +1,32 @@
+with Flow("Model Training Flow") as flow:
+    eval_metric = Parameter("Evaluation Metric", "auc")
+    df = get_data()
+    x, y, preprocessed_model = preprocessing(df)
```
preprocessed_model도 동일하게 mlflow에 log

Ilevk commented on train_prefect/tasks.py:
```diff
@@ -0,0 +1,168 @@
+    current_model = client.get_model_version(model_name, current_version)
+
+    name_filter = f"name='{model_name}'"
+    client = MlflowClient()
```
두번 정의할 필요가 없음

Ilevk commented on train_prefect/tasks.py:
```diff
@@ -0,0 +1,168 @@
+
+@task(log_stdout=True)
+def get_data():
+    url = "postgresql://postgres:postgres@localhost:5432/hoteldb"
```
접속 정보가 모두 드러난다 어떻게 숨길 수 있을까?

Ilevk commented on train_prefect/train_flow.py:
```diff
@@ -0,0 +1,32 @@
+    production_version = change_production_model(
+        model_name, current_version, eval_metric
+    )
+flow.run()
```
flow.register 해보기

lazychoi committed:
[feat-19] sqlalchemy create_url -> environment variable

tasks.py
- get_data()의 url에 postgres 비밀번호가 드러나는 것 수정
- => .bashrc 파일에 환경변수로 등록
- set_model() : xgboost, DecisionTree 중 선택하도록 설정

issue : #19

lazychoi committed:
[feat-19] Prefect Cloud 설정

train_flow.py
- flow.register(project_name="hotel")

issue : #19

AstroNaut000 reviewed (approved):
LGTM
